assets/js/modules/analytics-4/datastore/properties.test.js

```diff
@@ -1308,6 +1381,186 @@ describe( 'modules/analytics-4 properties', () => {
 			} );
 		} );
 
+		describe( 'getPropertyCreateTime', () => {
+			it( 'should use a resolver to make a network request', async () => {
+				fetchMock.get( propertyEndpoint, {
+					body: fixtures.properties[ 0 ],
+					status: 200,
+				} );
+
+				const propertyID = '12345';
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.setPropertyID( propertyID );
+
+				const initalPropertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( initalPropertyCreateTime ).toBeUndefined();
+
+				await untilResolved(
+					registry,
+					MODULES_ANALYTICS_4
+				).getPropertyCreateTime();
+				expect( fetchMock ).toHaveFetched( propertyEndpoint, {
+					query: { propertyID },
+				} );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toEqual(
+					new Date( fixtures.properties[ 0 ].createTime ).getTime()
+				);
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+			} );
+
+			it( 'should reset propertyCreateTime setting when property ID is PROPERTY_CREATE', async () => {
+				provideUserAuthentication( registry );
+
+				fetchMock.get( propertyEndpoint, {
+					body: {},
+					status: 200,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.selectProperty( PROPERTY_CREATE );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBe( '' );
+
+				expect( fetchMock ).toHaveFetchedTimes( 0 );
+			} );
+
+			it( 'should convert property time to unix timestamp when value is set', () => {
+				const createTimeInDateString = '2014-10-02T15:01:23Z';
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.setPropertyCreateTime( createTimeInDateString );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBe(
+					new Date( createTimeInDateString ).getTime()
+				);
+			} );
+
+			it( 'should reset create time when new property is created', async () => {
+				const accountID = fixtures.createProperty.parent;
+
+				fetchMock.post( createPropertyEndpoint, {
+					body: fixtures.createProperty,
+					status: 200,
+				} );
+
+				fetchMock.get( propertyEndpoint, {
+					body: {},
+					status: 200,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.createProperty( accountID );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBeUndefined();
+			} );
+
+			it( 'should reset create time when property is changed', async () => {
+				const propertyID = '1000';
+
+				const settings = {
+					propertyID: '12345',
+					webDataStreamID: '1000',
+					measurementID: 'abcd',
+					propertyCreateTime: 1662715085968,
+				};
+
+				provideUserAuthentication( registry );
```

Comment:
Yes, I included it to provide authentication to the store, as request is sent to it otherwise. As I recall `getProperty` needs user to be authenticated to make a request